### PR TITLE
interest accrue at LUP vs HTP

### PR DESCRIPTION
**Pull request by ith-harvey**
# Description of change
## High level
Updated how interest is accrued so as to always accrue at the lower of the LUP vs HTP

# Description of bug or vulnerability and solution
Updated how interest is accrued so as to always accrue at the lower of the LUP vs HTP

# Contract size
## Pre Change
```
============ Deployment Bytecode Sizes ============
 PoolCommons        -   8,762B  (35.65%)
```
## Post Change
```
============ Deployment Bytecode Sizes ============
PoolCommons              -   9,054B  (36.84%)
```
# Gas usage
Increase in gas as we hit Fenwick one more time to get current LUP when accruing interest
## Pre Change
```
| Function Name                        | min             | avg    | median | max    | # calls |
| addQuoteToken                        | 119315          | 175668 | 161115 | 694632 | 60004   |
| bucketTake                           | 327628          | 348558 | 348570 | 369464 | 4       |
| drawDebt                             | 248277          | 292957 | 268224 | 785846 | 136003  |
| kick                                 | 195615          | 221635 | 220800 | 986346 | 48000   |
| kickWithDeposit                      | 240830          | 275372 | 270174 | 983400 | 8000    |
| moveQuoteToken                       | 257319          | 257319 | 257319 | 257319 | 1       |
| removeQuoteToken                     | 133868          | 155338 | 158334 | 180868 | 4000    |
| repayDebt                            | 556944          | 589087 | 578983 | 737254 | 16002   |
| settle                               | 212455          | 212455 | 212455 | 212455 | 1       |
| take                                 | 97715           | 102846 | 102502 | 315779 | 15998   |
```

## Post Change
```
| Function Name                        | min             | avg    | median | max     | # calls |
| addQuoteToken                        | 119315          | 176540 | 161115 | 707453  | 60004   |
| bucketTake                           | 323735          | 344533 | 344473 | 365451  | 4       |
| drawDebt                             | 248277          | 293711 | 268224 | 798667  | 136003  |
| kick                                 | 196115          | 222377 | 221402 | 1031167 | 48000   |
| kickWithDeposit                      | 244973          | 279516 | 274317 | 996221  | 8000    |
| moveQuoteToken                       | 270679          | 270679 | 270679 | 270679  | 1       |
| removeQuoteToken                     | 145480          | 167395 | 171694 | 194228  | 4000    |
| repayDebt                            | 569765          | 601908 | 591804 | 750075  | 16002   |
| settle                               | 352711          | 352711 | 352711 | 352711  | 1       |
| take                                 | 97715           | 102844 | 102501 | 326762  | 15998   |
```

### CODE REVIEW

File: src/libraries/external/PoolCommons.sol [hunk 1]
```diff
@@ -221,8 +220,8 @@ library PoolCommons {
             // Scale the fenwick tree to update amount of debt owed to lenders
             Deposits.mult(
                 deposits_,
-                htpIndex,
-                (newInterest_ * 1e18) / depositAboveHtp + Maths.WAD // lender factor
+                accrualIndex,
+                Maths.floorWdiv(newInterest_, interestEarningDeposit) + Maths.WAD // lender factor
```
Comment (EdNoepel) [inline src/libraries/external/PoolCommons.sol:224]: Would prefer this function be named `Maths.wdivFloor`, but is outside scope of this PR.  No change in behavior here.

**Review by EdNoepel — APPROVED** [review]
In PR template, contract size section should show `PoolCommons` library, since that's the only one which changed.

Very happy with new variable names, layout of code, and commenting.  Looks really good.

**Review by mattcushman — APPROVED** [review]
lgtm!

**Review by mattcushman — APPROVED** [review]
lgtm!

File: src/libraries/external/PoolCommons.sol (hunk 1 above)
Comment (grandizzy) [reply src/libraries/external/PoolCommons.sol:224]: yep, will rename in follow up PR

**Comment by grandizzy** [comment]
> In PR template, contract size section should show `PoolCommons` library, since that's the only one which changed.
> 

updated

> Very happy with new variable names, layout of code, and commenting. Looks really good.

**Pull request merged**